src/ui/hooks/useMarkdown.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+
+import {YfmWrapper} from '../components/YfmWrapper/YfmWrapper';
+import {DL} from '../constants';
+import {getSdk} from '../libs/schematic-sdk';
+
+type Props = {
+    value: string;
+};
+
+const MarkdownCollection = new Map();
+async function renderMarkdown(value: string) {
+    if (!MarkdownCollection.has(value)) {
+        try {
+            const response = await getSdk().mix.renderMarkdown({text: value, lang: DL.USER_LANG});
+            const yfmString = response.result;
+            MarkdownCollection.set(value, yfmString);
+        } catch (e) {
+            console.error('useMarkdown failed ', e);
+        }
+    }
+
+    return MarkdownCollection.get(value);
+}
+
+export const useMarkdown = (props: Props) => {
+    const {value} = props;
+    const [markdown, setMarkdown] = React.useState<React.ReactNode>('');
```

Comment:
TBH it's not the best way to work with ReactNode.
I'd suggest to save in state result from `renderMarkdown(value)` as for ReactNode it's better to use ref or even memo

Reply:
> as for ReactNode it's better to use ref or even memo

This does not make much sense, since it is necessary to subscribe to the markdown value change


> I'd suggest to save in state result from renderMarkdown(value)

But it is possible to transfer the string to the state (and I did) - in this case, we duplicate the logic for converting the markup, but it seems that this is not so critical